src/app/slice/postSlice.js

```diff
@@ -0,0 +1,124 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { getAllPosts, getAllUsers, getPostById, getPostByUsername, getUserById } from "../../Helpers/Services/actions"; 
+// import { updateUser } from "../Auth/authSlice";
+
+const initialState = {
```

Comment:
you havent taken initial state ie post []

Reply:
managed using state.posts